airflow/bin/cli.py

```diff
@@ -80,6 +80,11 @@
 
 log = LoggingMixin().log
 
+DAGS_FOLDER = settings.DAGS_FOLDER
+
+if "BUILDING_AIRFLOW_DOCS" in os.environ:
+    DAGS_FOLDER = '[AIRFLOW_HOME]/dags'
```

Comment:
Could we get around all of these changes by instead setting `export AIRFLOW_HOME='$AIRFLOW_HOME'` before we build the docs?

Reply:
@ashb 

To answer your question, no setting `AIRFLOW_HOME` variable won't solve it as we take this value from `airflow.cfg`, but if we change `airflow.cfg` before building docs, this solves it for self-hosted docs but still fails for ReadTheDocs .

Reply:
Then perhaps `export AIRFLOW__CORE__DAGS_FOLDER=...` then

I feel that not having to change the code for this case just makes things a bit cleaner long term.

Reply:
Exporting `AIRFLOW__CORE__DAGS_FOLDER=` would mess up with ReadTheDocs environment as it would change the Dags folder and hence can break it when parsing some modules that use example dags.

We are already using this for `apply_defaults` decorator: 
https://github.com/apache/incubator-airflow/blob/f7fd78b06b24c15af3773b41ae3ecab7e48bb1ea/airflow/utils/decorators.py#L102

And also in future when we might have more such issues where docs would need to be handled differently, I think this would be better than setting environment variables for each instance.